lib/models/asset-size-printer.js

```diff
@@ -47,36 +47,30 @@ module.exports = class AssetPrinterSize {
     });
   }
 
-  makeAssetSizesObject() {
-    return new Promise((resolve) => {
-      let files = this.findFiles();
-      let testFileRegex = /(test-(loader|support))|(testem)/i;
+  async makeAssetSizesObject() {
+    let files = this.findFiles();
+    let testFileRegex = /(test-(loader|support))|(testem)/i;
 
-      // create a dedicated worker
-      const pool = workerpool.pool(`${__dirname}/worker.js`);
+    // create a dedicated worker
+    const pool = workerpool.pool(`${__dirname}/worker.js`);
 
-      let assets = files
-        // Skip test files
-        .filter((file) => {
-          let filename = path.basename(file);
-          return !testFileRegex.test(filename);
-        })
-        // Print human-readable file sizes (including gzipped)
-        .map((file) => {
-          return pool
-            .exec('gzipStats', [file])
-            .then((result) => result)
-            .catch((err) => console.error(err));
-        });
+    let assets = files
+      // Skip test files
+      .filter((file) => {
+        let filename = path.basename(file);
+        return !testFileRegex.test(filename);
+      })
+      // Print human-readable file sizes (including gzipped)
+      .map((file) => {
+        return pool
+          .exec('gzipStats', [file])
+          .then((result) => result)
+          .catch((err) => console.error(err));
+      });
```

Comment:
I think we'll want this in a try/finally block to ensure the pool is terminated:

```suggestion
    try {
      let assets = files
        // Skip test files
        .filter((file) => {
          let filename = path.basename(file);
          return !testFileRegex.test(filename);
        })
        // Print human-readable file sizes (including gzipped)
        .map((file) => {
          return pool
            .exec('gzipStats', [file])
            .catch((err) => console.error(err));
        });
        
        return await Promise.all(assets);
      } finally {
        pool.terminate();
      }
```